pkg/runctl/run_manager_test.go

```diff
@@ -1291,16 +1264,99 @@ func Test__runManager_Start__CreatesTektonTaskRun(t *testing.T) {
 	examinee.testing = newRunManagerTestingWithRequiredStubs()
 
 	// EXERCISE
-	resultError := examinee.Start(mockPipelineRun, config)
+	runNamespace, _, resultError := examinee.Start(mockPipelineRun, config)
 	assert.NilError(t, resultError)
 
 	// VERIFY
-	result, err := mockFactory.TektonV1beta1().TaskRuns(mockPipelineRun.GetRunNamespace()).Get(
+	result, err := mockFactory.TektonV1beta1().TaskRuns(runNamespace).Get(
 		tektonTaskRunName, metav1.GetOptions{})
 	assert.NilError(t, err)
 	assert.Assert(t, result != nil)
 }
 
+func Test__runManager_Start__Perform_cleanup_on_error(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range []struct {
+		name                     string
+		prepareRunNamespaceError error
+		createTektonTaskRunError error
+		cleanupError             error
+		failOnCleanupCount       int
+		expectedError            error
+		expectedCleanupCount     int
+	}{
+		{
+			name:                 "no failure",
+			expectedCleanupCount: 1, // before, no cleanup afterwards since no error occured
+		},
+		{
+			name:                     "failing inside prepareRunNamespace",
+			prepareRunNamespaceError: fmt.Errorf("cannot prepare run namespace: foo"),
```

Comment:
define the errors as variables in line 1279.

Reply:
done